spec/models/payment_spec.rb

```diff
@@ -0,0 +1,41 @@
+require "rails_helper"
+
+RSpec.describe Payment, type: :model do
+  describe "associations" do
+    it {should have_many(:tickets).dependent(:destroy)}
+    it {should belong_to(:user)}
+  end
+
+  describe "enum or status" do
+    it {should define_enum_for(:status).with_values(inactive: 0, active: 1)}
+  end
+
+  describe "scope" do
+    let!(:payment_1) {FactoryBot.create :payment, activated_at: "2022-08-22 08:39:57", status: :active}
+    let!(:payment_2) {FactoryBot.create :payment, activated_at: "2022-08-15 09:39:57", status: :inactive}
+    let!(:payment_3) {FactoryBot.create :payment, activated_at: "2022-09-02 07:39:57", status: :active}
```

Comment:
phân biệt cho c let vs let!

Reply:
Let và let! đều dùng để tạo data cho example
Let! sẽ được chạy khối before each ngầm cho mỗi example và giá trị của nó được cache lại trước khi example được thực thi.
Let sẽ chỉ được khởi tạo và tính toán khi gọi tới ở trong 1 example